### PR TITLE
Remove usage of go-ethereum library (adding fomichev/secp256k1)

### PR DESCRIPTION
**Pull request by timchurchard**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/Iotic-Labs/iotics-identity-go
 go 1.16
 
 require (
-	github.com/ethereum/go-ethereum v1.10.9
+	github.com/fomichev/secp256k1 v0.0.0-20180413221153-00116ff8c62f
 	github.com/go-bdd/gobdd v1.1.3
```
[the remark this answers is not in the document]
Comment (timchurchard) [reply go.mod:7]: gobdd v1.1.3 is the latest release and unrelated to my change. I'm resolving this comment.

File: pkg/register/helper.go [hunk 2]
```diff
@@ -14,5 +16,11 @@ import (
 // so it should probably be extracted to another common package
 func getPublicKeyFromBase58(publicBase58 string) (*ecdsa.PublicKey, error) {
 	publicKeyBytes := base58.DecodeAlphabet(publicBase58, base58.BTCAlphabet)
-	return ethcrypto.UnmarshalPubkey(publicKeyBytes)
+
+	curve := secp256k1.SECP256K1()
```
Comment (vtermanis) [inline pkg/register/helper.go:20]: I notice `proof/helper.go` has the same block. Is it worth factoring this unmarshal + make pub key bit at all?

**Review by vtermanis — COMMENTED** [review]
LGTM - just one question about common logic

File: pkg/register/helper.go (hunk 2 above)
Comment (timchurchard) [reply pkg/register/helper.go]: Thank you. I've refactored it to remove the two helper.go files. 0727f88

**Review by vtermanis — APPROVED** [review]
LGTM

**Pull request merged**